### PR TITLE
Automate building docker containers

### PR DESCRIPTION
**Pull request by enykeev**

### CODE REVIEW

File: .circle/docker.sh [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    sudo lxc-attach -n "$(docker inspect --format '{{.Id}}' ${2})" -- bash -c "${3}"
+  ;;
+  deploy)
+    if [ ${DEPLOY_DOCKER} -eq 0 ]; then
+      echo 'Skipping Docker push because DEPLOY_DOCKER=0'
+      exit
+    fi
+
+    for container in "${@:2}"; do
+      docker tag stackstorm/${container}:latest stackstorm/${container}:${ST2PKG_VERSION}
```
Comment (armab) [inline .circle/docker.sh]: Just to be sure, are you tagging Docker images `latest` if `ST2PKG_VERSION` is dev (ex. `1.4dev`)?
If not, what goes to latest then?

So here is an idea: 
* tag Docker `1.3.2` for stable versions like `1.3.2-5`
* tag Docker `latest` for unstable versions like `1.4dev-120`

So for `latest` tag we'll always have development versions, while every stable version is pinned via docker tag.

Comment (enykeev) [inline .circle/docker.sh]: Currently, I build "latest" for every container to ensure `docker-compose` would pick the latest version during test stage. It seem to me that latest should be only published for master no matter what version it currently is.

Comment (armab) [inline .circle/docker.sh]: `latest` in docker hub is pretty consistent thing, it doesn't works this way.

Ex. you can't tag & deploy `latest` for one version (ex. `1.4dev`) and during next build tag & overwrite `latest` for another version (ex. `1.3.2`).

Or users will deal with chaos.

Comment (enykeev) [inline .circle/docker.sh]: Absolutely. Or rather you can only move forward. Still makes perfect sense to me to publish it off a branch name\env variable. If you're worried that we may accidentally override 1.4dev latest with 1.3dev by reruning an older build, I don't see how we can avoid that short of simply exercising cautious.

**Comment by armab** [comment]
Regarding installing packages.
I'd suggest another way which wasn't possible with our old aka "local" approach.

Now when you triggering build after package is already in packagecloud - just add repo and `apt-get install` in Docker containers, instead of downloading & sharing artifact with Docker build.
It will kill a lot of nasty things we did before and we're doing now:
* no more `dpkg -i` install
* no more installing dependencies manually like `libssl-dev libffi-dev` which could change
* no `curl | jq` packagecloud


As a result: 
* for `master` branch: add `unstable` repo, install easily via `apt-get install`
* for `v1.3` branch: add `stable` repo, install with version pinning `1.3`
* finally follow `st2` branching

So with your new approach of triggering Docker build - it's possible to do it in more beautiful and straightforward way.

**Comment by enykeev** [comment]
Actually, I intentionally left `dpkg -i` because it doesn't require us to managing repositories and do multiple `apt-get update`\extra layers for the sake of convenience. Managing dependencies is an interesting point I didn't consider, though.

**Comment by armab** [comment]
Got it about upgrading, haven't thought about that case. Good point :+1: 

You can avoid package update with `apt` holding/pinning, see: https://help.ubuntu.com/community/PinningHowto#Introduction_to_Holding_Packages

**Pull request merged**